### PR TITLE
[sphinx] Fix path to source

### LINKED ISSUE #2398: 3.1.0: sphinx warnings `reference target not found`
First of all currently it is not possible to use straight `sphinx-build` command to build documentation out of source tree
<details>

```console
+ /usr/bin/sphinx-build -n -T -b man doc build/sphinx/man
Running Sphinx v7.2.6

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
  File "/home/tkloczko/rpmbuild/BUILD/astroid-3.1.0/doc/conf.py", line 41, in <module>
    from astroid.__pkginfo__ import __version__  # noqa
ModuleNotFoundError: No module named 'astroid'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/usr/lib/python3.9/site-packages/sphinx/application.py", line 211, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 181, in read
    namespace = eval_config_file(filename, tags)
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 371, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
  File "/home/tkloczko/rpmbuild/BUILD/astroid-3.1.0/doc/conf.py", line 41, in <module>
    from astroid.__pkginfo__ import __version__  # noqa
ModuleNotFoundError: No module named 'astroid'
+ /usr/bin/sphinx-build -n -T -b man doc build/sphinx/man
Running Sphinx v7.2.6

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
  File "/home/tkloczko/rpmbuild/BUILD/astroid-3.1.0/doc/conf.py", line 41, in <module>
    from astroid.__pkginfo__ import __version__  # noqa
ModuleNotFoundError: No module named 'astroid'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/usr/lib/python3.9/site-packages/sphinx/application.py", line 211, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 181, in read
    namespace = eval_config_file(filename, tags)
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 371, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
  File "/home/tkloczko/rpmbuild/BUILD/astroid-3.1.0/doc/conf.py", line 41, in <module>
    from astroid.__pkginfo__ import __version__  # noqa
ModuleNotFoundError: No module named 'astroid'
```
</details>

This can be fixed by patch like below:
```patch
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -21,7 +21,7 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath("../../"))
+sys.path.insert(0, os.path.abspath(".."))

 # -- General configuration -----------------------------------------------------
```

Please let me know if you want that patch as PR.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :scroll: Docs          |

## Description
This file was moved in
44f0065f7cd7224c917573fc7fc7a10c7d91e799.

Closes #2398

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -1 +1,2 @@
 add5f7b8eba427de9d39caae864bbc6dc37ef980
+2dd9027054db541871713ef1cb1ae89513d05555
```
Comment (Pierre-Sassoulas) [inline .git-blame-ignore-revs:2]:
```suggestion
# Apply black on doc/conf.py
2dd9027054db541871713ef1cb1ae89513d05555
```

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -1 +1,2 @@
 add5f7b8eba427de9d39caae864bbc6dc37ef980
```
Comment (Pierre-Sassoulas) [inline .git-blame-ignore-revs:1]:
```suggestion
# Initial formatting of astroid 
add5f7b8eba427de9d39caae864bbc6dc37ef980
```

**Review by Pierre-Sassoulas — APPROVED** [review]

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -1 +1,4 @@
+# Initial formatting of astroid
 add5f7b8eba427de9d39caae864bbc6dc37ef980
+# Apply black on doc/conf.py
+2dd9027054db541871713ef1cb1ae89513d05555
```
Comment (Pierre-Sassoulas) [inline .git-blame-ignore-revs:4]:
```suggestion
2dd9027054db541871713ef1cb1ae89513d05555
# Black's 2024 style
396f01a15d1cb0351b33654acdeedde64f537a0c
```

Comment (Pierre-Sassoulas) [reply .git-blame-ignore-revs:4]: We missed other big formatting change tbh, but maybe not worth it to track them down.

Comment (Pierre-Sassoulas) [reply .git-blame-ignore-revs:4]: * | 396f01a1 - [style] Apply black 2024's style (#2367) (3 months ago) <pre-commit-ci[bot]>

* | 0942d8fa - Bump black from 23.1a1 to 23.1.0 (#2011) (1 year, 3 months ago) <dependabot[bot]>

* | 380e8861 - [pre-commit] Trying out black 23.1a1 (#1909) (1 year, 4 months ago) <Pierre Sassoulas>

* | 344454c8 - Bump black from 22.10.0 to 22.12.0 (#1900) (1 year, 5 months ago) <dependabot[bot]>

* | b0220bdd - Bump black from 22.8.0 to 22.10.0 (#1825) (1 year, 7 months ago) <dependabot[bot]>

* | 731d0515 - Apply black on the mechanize brain (1 year, 7 months ago) <Pierre Sassoulas>

* | 3a0b104e - Bump black from 22.6.0 to 22.8.0 (1 year, 8 months ago) <dependabot[bot]>

* | 88ccd6fd - Bump black from 22.3.0 to 22.6.0 (#1672) (1 year, 10 months ago) <dependabot[bot]>

| * 264357f5 - Bump black from 22.1.0 to 22.3.0 (#1496) (2 years, 1 month ago) <dependabot[bot]>

* | 3ecb96dc - Bump black from 22.1.0 to 22.3.0 (#1496) (2 years, 1 month ago) <dependabot[bot]>

* 2dd90270 - Apply black on doc/conf.py too (2 years, 10 months ago) <Pierre Sassoulas>

| * 7409caf9 - Reformats according to black (3 years, 2 months ago) <hippo91>

* 40373681 - Fix black issues (3 years, 2 months ago) <Marc Mueller>

* b20cc9c5 - Update black version - tox (3 years, 2 months ago) <Marc Mueller>

* 0d8ed3f8 - pylint + black formatting (3 years, 3 months ago) <hippo91>

| * | | | | | 01c678ad - Formatting according to black (3 years, 3 months ago) <hippo91>

| * | | | | f1613f46 - Reformated according to black (3 years, 4 months ago) <hippo91>

| | * | | | 3554cab5 - Formatting according to black (3 years, 4 months ago) <hippo91>

| * | | | | a324fbdf - Formatting according to black (3 years, 4 months ago) <hippo91>

| * | | | dbe56fcc - Formatting according to black (3 years, 5 months ago) <hippo91>

* 18c99c4e - Reformatting according to black (4 years, 5 months ago) <hippo91>

| * | | | 4ab338ed - Refomat according to black (5 years ago) <hippo91>

| | * eeae266d - Reformat according to black (5 years ago) <hippo91>

* 3d58a62c - Let formatting be handled by black (6 years ago) <Claudiu Popa>

* c6581e98 - Add black step in tox.ini and Travis, as well as the badge (6 years ago) <Claudiu Popa>

Comment (jacobtylerwalls) [reply .git-blame-ignore-revs:4]: Yeah when I started with one hash I wasn't trying to get em all 😉

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**